vignettes/rcppannoy.Rmd

```diff
@@ -10,6 +10,13 @@ vignette: >
   %\usepackage[utf8]{inputenc}
 ---
 
+```{r, echo=FALSE, results="hide"}
+knitr::opts_chunk$set(error=FALSE, warning=FALSE, message=FALSE)
```

Comment:
You could set `eval=FALSE` globally too.  

Brooke Anderson and I used that super-cleverly in the R Journal paper on using drat for a data archive -- if the optional data is not there, the knitr chunks with not evaluate.  After some pretest for a package we essentially do

```
# Code to add at the beginning of the vignette(s) of the code package to ensure vignette
# code is conditioned to only run if the data package is available
hasData <- requireNamespace("hurricaneexposuredata", quietly = TRUE)                #1
if (!hasData) {                                                                     #2
  knitr::opts_chunk$set(eval = FALSE)                                             #3
  msg <- paste("Note: Examples in this vignette require that the",
               "`hurricaneexposuredata` package be installed. The system",
               "currently running this vignette does not have that package",
               "installed, so code examples will not be evaluated.")
  msg <- paste(strwrap(msg), collapse="\n")
  message(msg)                                                                    #4
}
```

I think we discuss the trick in the paper too -- the hashmarks one to four here are references.

Reply:
Yeah, that's probably neater than having to set `eval=FALSE` manually.